crash in put

Repro is with Zed commit 59dc184.

This should be easy to fix...

```
bay 3312 > cat bug.zson
{
    commit: 0x0dbd4686a5f476269b7c056b24b6bb72f3074b6b (=ksuid.KSUID),
    segment: {
        id: 0x0dbd4686be2d193332b7bff2896f7d8e401c8455 (ksuid.KSUID),
        meta: {
            first: null,
            last: null,
            count: 1 (uint64),
            row_size: 10
        } (=segment.Metadata)
    } (=segment.Reference)
} (=commits.Add)
bay 3313 > zq "put segment.foo:=0" bug.zson
panic: interface conversion: zng.Type is *zng.TypeAlias, not *zng.TypeRecord
goroutine 35 [running]:
runtime/debug.Stack(0x10061646174654d, 0x8005268, 0x0)
    /usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/brimdata/zed/zqe.RecoverError(0x17e12e0, 0xc00043c5d0, 0xc00043c5d0, 0x41)
    /Users/mccanne/repo/zed/zqe/error.go:163 +0x26
github.com/brimdata/zed/driver.safePull.func1(0xc0000cbf38)
    /Users/mccanne/repo/zed/driver/driver.go:187 +0x4b
panic(0x17e12e0, 0xc00043c5d0)
    /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/brimdata/zed/proc/put.(*Proc).deriveRecordSteps(0xc000323ef0, 0xc0000cba10, 0x0, 0x0, 0xc000339c80, 0x2, 0x2, 0xc0003e3b60, 0x1, 0x1, ...)
    /Users/mccanne/repo/zed/proc/put/put.go:256 +0x19d6
github.com/brimdata/zed/proc/put.(*Proc).deriveSteps(0xc000323ef0, 0xc00043c480, 0xc0003e3b60, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    /Users/mccanne/repo/zed/proc/put/put.go:224 +0x230
github.com/brimdata/zed/proc/put.(*Proc).lookupRule(0xc000323ef0, 0xc00043c480, 0xc0003e3b60, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    /Users/mccanne/repo/zed/proc/put/put.go:336 +0x1bb
github.com/brimdata/zed/proc/put.(*Proc).put(0xc000323ef0, 0xc000339cc0, 0xc000339cc0, 0xc000336310, 0x0)
    /Users/mccanne/repo/zed/proc/put/put.go:352 +0x136
github.com/brimdata/zed/proc/put.(*Proc).Pull(0xc000323ef0, 0x0, 0x0, 0x0, 0x0)
    /Users/mccanne/repo/zed/proc/put/put.go:372 +0x165
github.com/brimdata/zed/driver.safePull(0x800d5d8, 0xc000323ef0, 0x0, 0x0, 0x0, 0x0)
    /Users/mccanne/repo/zed/driver/driver.go:189 +0x6a
github.com/brimdata/zed/driver.run.func1(0x800d5d8, 0xc000323ef0, 0xc0003f61e0, 0xc0003e2e40)
    /Users/mccanne/repo/zed/driver/driver.go:108 +0x4d
created by github.com/brimdata/zed/driver.run
    /Users/mccanne/repo/zed/driver/driver.go:106 +0x165
```

--- a/proc/put/put.go
+++ b/proc/put/put.go
@@ -253,7 +253,7 @@ func (p *Proc) deriveRecordSteps(parentPath field.Path, inCols []zng.Column, val
 			cols = append(cols, zng.Column{inCol.Name, vals[matchIndex].Type})
 		// input record field overwritten by nested assignment: recurse.
 		case len(path) < len(matchPath) && zng.IsRecordType(inCol.Type):
-			nestedStep, typ, err := p.deriveRecordSteps(path, inCol.Type.(*zng.TypeRecord).Columns, vals)
+			nestedStep, typ, err := p.deriveRecordSteps(path, zng.TypeRecordOf(inCol.Type).Columns, vals)
 			if err != nil {
 				return step{}, nil, err
 			}

--- a/proc/put/ztests/nested.yaml
+++ b/proc/put/ztests/nested.yaml
@@ -6,6 +6,7 @@ inputs:
       {}
       {"a": null}
       {"a": {}}
+      {"a": {}(=alias)}
       {"a": {"c": null}}
       {"a": {"c": {}}}
       {"a": {"c": {"a": 4}}}
@@ -17,6 +18,7 @@ outputs:
   - name: stdout
     data: |
       {pre:"pre",a:{a:1,b:2,c:{a:3},d:4},b:5,post:"post"}
+      {a:{a:1,b:2,c:{a:3},d:4},pre:"pre",b:5,post:"post"}
       {a:{a:1,b:2,c:{a:3},d:4},pre:"pre",b:5,post:"post"}
       {a:{a:1,b:2,c:{a:3},d:4},pre:"pre",b:5,post:"post"}
       {a:{c:{a:3},a:1,b:2,d:4},pre:"pre",b:5,post:"post"}

